File: 14-Longest-Common-Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */








var longestCommonPrefix = function(strs) {
    let newStr = ''
    let commonPrefix = strs[0]

    for (let i = 0; i < commonPrefix.length; i++){
        let char =commonPrefix[i]
       for (let j = 1; j < strs.length; j++){
            if(char !== strs[j][i]) return newStr
        } 
        newStr += char
    }
    return newStr
}

File: 14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        newStr = ''
        commonPrefix = strs[0]

        for i in range(len(commonPrefix)):
            char = commonPrefix[i]
            for j in range(1,len(strs)):
                if i >= len(strs[j]) or char != strs[j][i]:
                    return newStr
            newStr += char
        
        return newStr



        